src/Sentry/Internal/MainExceptionProcessor.cs

```diff
@@ -155,7 +155,7 @@ private SentryException BuildSentryException(Exception exception, int id, int? p
             sentryEx.Mechanism = mechanism;
         }
 
-        sentryEx.Stacktrace = SentryStackTraceFactoryAccessor().Create(exception);
+        sentryEx.Stacktrace ??= SentryStackTraceFactoryAccessor().Create(exception);
```

Comment:
What's the implication here? Previously, the stacktrace would get overwritten. Why is this change necessary? Where would the stacktrace come from?

Reply:
the stack trace is already set in the event. We get it from the blocking detection logic so we won't want it overwriten

Reply:
```suggestion
        sentryEx.Stacktrace = SentryStackTraceFactoryAccessor().Create(exception);
```

The `sentryEx` gets newly instantiated in this method and the `StackTrace` has to be set explicitly. I think this change does nothing.

> We get it from the blocking detection logic so we won't want it overwriten

That should all happen in the event processor.